src/tir/schedule/state.cc

```diff
@@ -421,6 +389,86 @@ class StateCreator : private StmtVisitor {
   arith::Analyzer analyzer_;
 };
 
+/*! \brief A helper class to create a new ScheduleStateNode from an IRModule */
+class StateCreator : private StmtVisitor {
```

Comment:
I just checked. Actually StateCreator, including its contents like PushSRef, VisitStmt, is not used anymore, so let's remove them all and inline the Create method into `ScheduleState::ScheduleState`

Reply:
Update: I'm not sure I'm 100% correct after a second check...Let me know :-)